Blog App - processing data in models

ragangithub opened the pull request:
- Created User,Post,Comment,and Like model
- Created a method that returns the 3 most recent posts for a given user.
- Created a method that updates the posts counter for a user.
- Created a method which returns the 5 most recent comments for a given post.
- Created a method that updates the comments counter for a post.
- Created a method that updates the likes counter for a post.

ragangithub committed:
Add model

ragangithub committed:
Fix linter

mckent05 commented on app/models/comment.rb:
```diff
@@ -0,0 +1,8 @@
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_comment_count
+    post.increment!(:comments_counter)
+  end
```
- Good job done so far, but according to the project requirements, I suggest you call this function after the data has been saved to your database. Please fix this 🙏 i.e `after_save :update_comment_counter`

mckent05 commented on app/models/like.rb:
```diff
@@ -0,0 +1,8 @@
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_like_count
+    post.increment!(:likes_counter)
+  end
```
- Good job done so far, but according to the project requirements, I suggest you call this function after the data has been saved to your database. Please fix this 🙏 i.e `after_save :update_like_count`

mckent05 commented on app/models/post.rb:
```diff
@@ -0,0 +1,13 @@
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def update_posts_count
+    author.increment!(:posts_counter)
+  end
```
- Good job done so far, but according to the project requirements, I suggest you call this function after the data has been saved to your database. Please fix this 🙏 i.e `after_save :update_posts_count`

mckent05 reviewed (changes requested):
Hi @ragangithub, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
![almost there](https://user-images.githubusercontent.com/73607512/207265056-d056ea78-77c5-4b66-802e-694a5ed4ecf3.gif)

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job setting up the associations between the models 👏 👏 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ragangithub committed:
Update comment like and post models

ragangithub commented:
- Added after_save :update_posts_count for post model
- Added after_save :update_like_count for like model
- Added after_save :update_comment_counter for comment model

cynthiainga commented on app/models/comment.rb:
```diff
@@ -0,0 +1,10 @@
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  after_save :update_comment_count
+
+  def update_comment_count
+    post.increment!(:comments_counter)
+  end
```
- [Optional] Good job implementing the requested changes, after you should make the `update_comment_count` method private 🙃 . Kindly do the same for `update_like_count` and `update_posts_count` methods 😁

cynthiainga reviewed (approved):
Hi @ragangithub, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

![](https://user-images.githubusercontent.com/69011963/151541305-67f4f8e4-c79e-4f9d-a6cf-2c7efc3102c6.gif)

## Highlights

- [ ] No linters errors ✔️ 
- [ ] Correct Gitflow ✔️ 
- [ ] Good job setting up the associations between the models ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
